src/libs/actions/Report.js

```diff
@@ -849,6 +849,20 @@ function handleReportChanged(report) {
         return;
     }
 
+    // It is possible that we optimistically created a DM/group-DM for a set of users for which a report already exists.
+    // In this case, the API will let us know by returning a preexistingReportID.
+    // We should clear out the optimistically created report and re-route the user to the preexisting report.
```

Comment:
Where does this happen: `We should clear out the optimistically created report and re-route the user to the preexisting report.`? I've made use of the `preexistingReportID` to navigate the user to an existing report after creating a money request, but the stale optimistic report does not get removed for me. I had to add `Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${report?.reportID}`, null);
` after the navigation line, cc @pasyukevich @blazejkustra

Reply:
Hm, good question. I'm not seeing it either. Maybe we forgot to do it?